### ISSUE TITLE
Architecture of osquery

### ISSUE THREAD

**Issue by vasiqmz** [facebook/osquery#2796]
Hi everyone,

After going through the documentation of osquery from the site and personally using the project, it's amazing. 
I want to dive further into the coding part of the project and understand more about its architecture and it's various components.
Any help would be appreciated.
Thank you.

**Comment by theopolis** [comment]
Hi @vasiqmz! Thanks for the kind words about the project.

Have you gone through the building documentation: http://osquery.readthedocs.io/en/stable/development/building/

That entire `Development` subsection may be slightly outdated, but will serve as a bootstrap for understanding how the project is laid out. After that the project is fairly simple, is there an area you're interested in improving?

**Comment by vasiqmz** [comment]
Currently I'm trying to understand the inner-working of osquery - how does it work, the modules they are dependent on, how does it interact with the OS, how the dynamic tables are generated, etc. I know those sort of questions can only be answered when you dive into the code but I thought I better ask here first.

Any advise on it ?

And yes, I had gone through the Development subsection but it was mainly about building osquery from scratch.

**Comment by theopolis** [comment]
That link I provided shows you how to build from scratch. :) And there's a page on exactly how the tables are generated too!

I'd recommend trying to build the project, following the building documentation I linked, then going through some of the code in `./core` and `./sql`.

> Any advise on it ?

I'm not sure what you're looking for specifically, how about jumping onto the osquery Slack discussion and let us know where you get stuck: https://osquery-slack.herokuapp.com/
